pandas/core/strings/accessor.py

```diff
@@ -2482,12 +2482,16 @@ def get_dummies(
         1   False  False   False
         2   True   False   True
         """
+        from pandas.core.dtypes.common import is_string_dtype
+
         from pandas.core.frame import DataFrame
 
+        if is_string_dtype(dtype):
+            raise ValueError("string dtype not supported, please use a numeric dtype")
```

Comment:
I wonder if instead of explicitly disallowing specifically string dtype, should we allow any numeric dtype? (so raise an error `if not is_numeric_dtype(dtype)`) 
Because right now this would also allow the user to specify they want datetime dtype, for example, which also does not make much sense?